addons/binding/org.openhab.binding.zwave/src/main/java/org/openhab/binding/zwave/internal/protocol/commandclass/ZWaveAlarmCommandClass.java

```diff
@@ -445,6 +496,14 @@ public Integer getValue() {
                 result.add(getSupportedMessage());
             }
         }
+        if (getVersion() > 2) {
```

Comment:
I think this should be moved up into the 'refresh==true' check. The problem is that this could end up in a continuous loop since this will be called multiple times, and each time you might add more messages if the device doesn't report any events. I don't know how likely this is - knowing how bad some implementations are, it's likely, so we should make sure that there's something to get us through this initialisation stage.